src/Concerns/HasContent.php

```diff
@@ -0,0 +1,179 @@
+            return $this->contents()->createMany($renderable->map(function (Renderable $r) {
+                return $this->formatKeys($r);
+            }));
+        }
+
+        $attach = array_merge($this->formatKeys($renderable), $identifier ? ['identifier' => $identifier] : []);
+
+        return collect([$this->contents()->create($attach)]);
+    }
+
+    /**
+     * Update the models attached via the contents() relation to match the passed collection of $renderables.
+     *
+     * @param  Renderable|Collection|array  $renderables either a single instance of Renderable or a collection/array of them.
+     * @return array[]
+     */
+    public function syncContent(Renderable|Collection|array $renderables, bool $detaching = true): array
+    {
+        $changes = [
+            'attached' => [], 'detached' => [],
+        ];
+
+        // get intersect of input and attached
+        // Diff on collection of models only works when both collections are an Eloquent Collection.
+        $renderables = EloquentCollection::wrap($renderables);
+        $attached = EloquentCollection::make($this->contents->pluck('renderable'));
+        $intersect = $renderables->intersect($attached);
+
+        // diff intersect from attached --> this gives detaching
+        if ($detaching) {
+            $detach = $attached->diff($intersect);
+
+            if (count($detach) > 0) {
+                $this->detachContent($detach);
+                $changes['detached'] = $this->formatKeys($detach);
+            }
+        }
+
+        // diff intersect from input --> this gives attaching
+        $attach = $renderables->diff($intersect);
+        if (count($attach) > 0) {
+            $changes['attached'] = $this->formatKeys($this->attachContent($attach)->pluck('renderable'));
+        }
+
+        if (count($changes['attached']) + count($changes['detached']) > 0) {
+            $this->touch();
+        }
+
+        return $changes;
+    }
+
+    /**
+     * Remove passed renderables from this contentables contents() relation.
+     *
+     * @param  Renderable|Collection|array  $renderables either a single instance of Renderable or a collection/array of them.
+     */
+    public function detachContent(Renderable|Collection|array $renderables): void
+    {
+        $contentModel = config('contentable.model');
+
+        $renderables = EloquentCollection::wrap($renderables);
+
+        $classes = [];
+        foreach ($renderables as $renderable) {
+            $classes[$renderable::class][] = $renderable->getKey();
+        }
+
+        foreach ($classes as $class => $ids) {
+            $contentModel::where('contentable_id', $this->getKey())
+                ->where('contentable_type', self::class)
+                ->where('renderable_type', $class)
+                ->whereIn('renderable_id', $ids)
+                ->delete();
+        }
+    }
+
+    private function formatKeys(Renderable|Collection|array $renderables)
```

Comment:
One small thing which can be ignored if you choose. But the typingting technically needs to change or the interface needs to add `getKey()`. I would opt for changing the typehint.

You also would need to import the Model class if you opt to make this change. I will add a suggestion above which you can choose to commit or ignore along with this.

```suggestion
    private function formatKeys((Model&Renderable)|Collection|array $renderables)
```